src/composables/accounts.ts

```diff
@@ -222,12 +241,95 @@ export function useAccounts() {
     setActiveAccountByProtocolAndIdx(protocol, account?.idx || 0);
   }
 
-  function setMnemonic(newMnemonic: string) {
+  /**
+   * Setting/Resetting the password key logs the user in/out.
+   */
+  function setPasswordKey(key: IKey | null) {
+    passwordKey.value = key;
+    if (IS_EXTENSION) {
+      if (key) {
+        startSession(key, secureLoginTimeout.value);
+      } else {
+        endSession();
+      }
+    }
+  }
+
+  async function openLoginModal() {
+    setLoaderVisible(true);
+    const sessionKey = await getSessionKey();
+    if (sessionKey) {
+      setPasswordKey(sessionKey);
+      setLoaderVisible(false);
+      return;
+    }
+    setLoaderVisible(false);
+
+    // If the tab is offscreen, we need to wait for the session key to be set
+    // Maybe it makes sense to open a popup and ask for the password
+    if (IS_OFFSCREEN_TAB) {
+      await new Promise<void>((resolve) => {
+        const interval = setInterval(async () => {
+          const key = await getSessionKey();
+          if (key) {
+            setPasswordKey(key);
+            clearInterval(interval);
+            resolve();
+          }
+        }, 5000);
+      });
+      return;
+    }
+    const { openModal } = useModals();
```

Comment:
I suggest to move this to the beginning of the composable to match the common practice we use.

Reply:
Initializing it at the beginning of the composable wont work in this case